github/store/db.go

```diff
@@ -47,72 +47,180 @@ const (
+	SELECT owner_login AS owner, name, full_name, private, description
+	FROM github_repositories_versioned WHERE %v = ANY(versions)`, v))
+	if err != nil {
+		return fmt.Errorf("failed to create unified view repositories: %v", err)
+	}
+
+	_, err = s.DB.ExecContext(ctx, "DROP MATERIALIZED VIEW IF EXISTS issues")
+	if err != nil {
+		return fmt.Errorf("failed to drop unified view issues: %v", err)
+	}
+	_, err = s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE MATERIALIZED VIEW issues AS
+	SELECT repository_owner, repository_name, repository_owner || '/' || repository_name AS repository_full_name,
+           number, state, title, body,
+           created_at, closed_at, updated_at, comments, user_id, user_login, htmlurl AS html_url, labels
+	FROM github_issues_versioned WHERE %v = ANY(versions)`, v))
+	if err != nil {
+		return fmt.Errorf("failed to create unified view issues: %v", err)
+	}
+
+	_, err = s.DB.ExecContext(ctx, "DROP MATERIALIZED VIEW IF EXISTS issue_comments")
+	if err != nil {
+		return fmt.Errorf("failed to drop unified view issue_comments: %v", err)
+	}
+	_, err = s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE MATERIALIZED VIEW issue_comments AS
+	SELECT c.repository_owner, c.repository_name, c.repository_owner || '/' || c.repository_name AS repository_full_name,
+	       c.issue_number, c.created_at, c.body, c.user_id, c.user_login, c.htmlurl AS html_url
+	FROM github_issue_comments_versioned AS c
+	JOIN github_issues_versioned AS i ON
+		 i.repository_owner = c.repository_owner AND
+		 i.repository_name = c.repository_name AND
+		 i.number = c.issue_number WHERE %v = ANY(c.versions)`, v))
+	if err != nil {
+		return fmt.Errorf("failed to create unified view issue_comments: %v", err)
+	}
+
+	_, err = s.DB.ExecContext(ctx, "DROP MATERIALIZED VIEW IF EXISTS pull_requests")
+	if err != nil {
+		return fmt.Errorf("failed to drop unified view pull_requests: %v", err)
+	}
+	_, err = s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE MATERIALIZED VIEW pull_requests AS
+	SELECT repository_owner, repository_name, repository_owner || '/' || repository_name AS repository_full_name,
+	       number, state, title, body, created_at, closed_at, merged_at, updated_at,
+	       commits, comments, changed_files, additions, deletions, review_comments AS reviews,
+	       user_id, user_login, base_repository_name, base_repository_owner,
+	       base_repository_name || '/' || base_repository_owner AS base_repository_full_name,
+	       head_ref, head_sha, merge_commit_sha, htmlurl AS html_url, labels
+	FROM github_pull_requests_versioned WHERE %v = ANY(versions)`, v))
+	if err != nil {
+		return fmt.Errorf("failed to create unified view pull_requests: %v", err)
+	}
+
+	_, err = s.DB.ExecContext(ctx, "DROP MATERIALIZED VIEW IF EXISTS pull_request_reviews")
+	if err != nil {
+		return fmt.Errorf("failed to drop unified view pull_request_reviews: %v", err)
+	}
+	// TODO: state value rename COMMENTED to something else?
```

Comment:
makes sense to fix it here. I just want to hear somebody else's opinion on the new name.
Current statuses are approved/rejected/commented. I would rename it to "viewed" (in one of bitbuckets review actually have a boolean variable "approved", which would map nicely on "viewed" but not "commented")

Reply:
I don't see the benefit of "viewed", because I'd say that the purpose of a review is either approve, reject or comment the PR.
Also, I didn't understand you when mentioning BB reviews; but if it means that BB reviews can only happen to approve or not, I'd map them as `approved`/`commented`

Reply:
we can keep "commented" if you prefer.

Reply:
not a strong opinion, for sure, but I'd vote for `comment` if you ask ;)

Reply:
@smacker could you please recap the values for the 3 different providers?

Reply:
github: APPROVED/CHANGES_REQUESTED/COMMENTED
bbserver: APPROVED/REVIEWED
for bbcould&gitlab there is a boolean field "approved" (according to the table) but maybe if we fetch data from activity endpoints we can get some text values.

Reply:
If have to go for a duality, I'd go for `APPROVED/COMMENTED`. Simply because a comment could also not be a review, but a review should always be a comment.

Reply:
`APPROVED/COMMENTED` sounds good to me.

Reply:
fixed